src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Search/SearchOptionsFactoryTests.cs

```diff
@@ -349,6 +354,26 @@ public void GivenInvalidCompartmentId_WhenCreated_ThenExceptionShouldBeThrown(st
             Assert.Equal("Compartment id is null or empty.", exception.Message);
         }
 
+        [Theory]
+        [InlineData(TotalType.Accurate)]
+        [InlineData(TotalType.None)]
+        public void GivenNoTotalParameter_WhenCreated_ThenDefaultSearchOptionsShouldHaveCountWhenConfiguredByDefault(TotalType type)
+        {
+            _coreFeatures.IncludeTotalInBundle = type;
+
+            SearchOptions options = CreateSearchOptions(queryParameters: null);
+
+            Assert.Equal(type, options.IncludeTotal);
+        }
+
+        [Fact]
+        public void GivenNoTotalParameterWithInvalidDefault_WhenCreated_ThenDefaultSearchOptionsThrowException()
```

Comment:
Would it be useful to add a test that has `total` in the query parameters and validate that it replaces the default value?

Reply:
Good idea